core/src/test/java/org/apache/calcite/util/UtilTest.java

```diff
@@ -1560,30 +1560,6 @@ private void checkCompositeMap(String[] beatles, Map<String, Integer> map) {
     assertThat(Util.distinctList(cbcaC), is(Arrays.asList("c", "b", "a")));
   }
 
-  /** Unit test for {@link Utilities#hashCode(double)}. */
-  @Test public void testHash() {
-    checkHash(0d);
-    checkHash(1d);
-    checkHash(-2.5d);
-    checkHash(10d / 3d);
-    checkHash(Double.NEGATIVE_INFINITY);
-    checkHash(Double.POSITIVE_INFINITY);
-    checkHash(Double.MAX_VALUE);
-    checkHash(Double.MIN_VALUE);
-  }
-
-  public void checkHash(double v) {
-    assertThat(Double.valueOf(v).hashCode(), is(Utilities.hashCode(v)));
-    final long long_ = (long) v;
-    assertThat(Long.valueOf(long_).hashCode(), is(Utilities.hashCode(long_)));
-    final float float_ = (float) v;
-    assertThat(Float.valueOf(float_).hashCode(),
-        is(Utilities.hashCode(float_)));
-    final boolean boolean_ = v != 0;
-    assertThat(Boolean.valueOf(boolean_).hashCode(),
-        is(Utilities.hashCode(boolean_)));
-  }
```

Comment:
Although this is deprecated (which I think makes sense), shouldn't the tests stay until the code is actually removed?

Reply:
I thought of that, however
1) Implementation of `Utilities.hashCode(boolean/int/...)` just delegates to the relevant `Boolean.hashCode` methods. Does it really make sense to test Java implementation? 
2) If I keep the tests, it would result in deprecated methods being used which would raise compiler warnings.

Do you still think the test should be there?